pkg/idp/provider/azure/request.go

```diff
@@ -27,12 +28,63 @@ func ProviderScope() []string {
 	return []string{"https://graph.microsoft.com/.default"}
 }
 
-func getPermissionCreateRequestBody(parentApp models.Applicationable) models.Applicationable {
-	app := models.NewApplication()
-	app.SetRequiredResourceAccess(parentApp.GetRequiredResourceAccess())
-	return app
+func computeAppUpdatePatch(config provider.AppConfig, app models.Applicationable, parentApp models.Applicationable) (bool, models.Applicationable) {
```

Comment:
This function computes a patch for the app by comparing desired and current state for each property